src/commands/extract.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2022 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use std::{fs, path::PathBuf};
+
+use anyhow::{bail, Result};
+use c2pa::{jumbf_io, Ingredient, ManifestStore};
+use clap::Parser;
+
+use crate::commands::{load_trust_settings, Trust};
+
+#[derive(Debug, Parser)]
+pub enum Extract {
+    /// Extract the .json or .c2pa manifest.
+    Manifest {
+        /// Input path to asset.
+        path: PathBuf,
+
+        /// Path to output file.
+        #[clap(short, long)]
+        output: PathBuf,
+
+        /// Extract binary .c2pa manifest.
+        #[clap(short, long)]
+        binary: bool,
+
+        /// Force overwrite output if it already exists.
+        #[clap(short, long)]
+        force: bool,
+
+        #[clap(flatten)]
+        trust: Trust,
+    },
+    /// Extract the .json ingredient.
+    Ingredient {
+        /// Input path to asset.
+        path: PathBuf,
+
+        /// Path to output ingredient .json.
+        #[clap(short, long)]
+        output: PathBuf,
+
+        /// Force overwrite output if it already exists.
+        #[clap(short, long)]
+        force: bool,
+
+        #[clap(flatten)]
+        trust: Trust,
+    },
+    /// Extract known resources from a manifest (e.g. thumbnails).
+    Resources {
+        /// Input path(s) to asset(s).
+        paths: Vec<PathBuf>,
+
+        /// Path to output folder.
+        #[clap(short, long)]
+        output: PathBuf,
+
+        /// Force overwrite output and clear children if it already exists.
+        #[clap(short, long)]
+        force: bool,
+
+        #[clap(flatten)]
+        trust: Trust,
+        //
+        // TODO: add flag for additionally exporting unknown ingredients (ingredients that
+        // do not have a standardized label) as a binary file
+    },
+}
+
+impl Extract {
+    pub fn execute(&self) -> Result<()> {
+        match self {
+            Extract::Manifest {
+                path,
+                output,
+                binary,
+                force,
+                trust,
+            } => {
+                if !path.exists() {
+                    bail!("Input path does not exist")
```

Comment:
That's a good idea. Another option is to introduce an argument such as `--format`, similar to cargo's `--message-format=json`. It can provide detailed information about command success/failure, e.g. paths that successfully signed and paths that failed to sign. I think this would pair well with exit codes?

Here's a real example of how cargo outputs with `--message-format=json` (there are multiple outputs like this as the command runs):
```
{
   "reason":"compiler-artifact",
   "package_id":"registry+https://github.com/rust-lang/crates.io-index#mockall@0.12.1",
   "manifest_path":"/Users/me/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mockall-0.12.1/Cargo.toml",
   "target":{
      "kind":[
         "lib"
      ],
      "crate_types":[
         "lib"
      ],
      "name":"mockall",
      "src_path":"/Users/me/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mockall-0.12.1/src/lib.rs",
      "edition":"2021",
      "doc":true,
      "doctest":true,
      "test":true
   },
   "profile":{
      "opt_level":"0",
      "debuginfo":2,
      "debug_assertions":true,
      "overflow_checks":true,
      "test":false
   },
   "features":[
      
   ],
   "filenames":[
      "/Users/me/Documents/repos/c2patool/target/debug/deps/libmockall-7bdbc60eb0277649.rlib",
      "/Users/me/Documents/repos/c2patool/target/debug/deps/libmockall-7bdbc60eb0277649.rmeta"
   ],
   "executable":null,
   "fresh":true
}
```

Reply:
Also, it may help to restructure the program to remove its dependence on `anyhow`. If we provide detailed errors we can support detailed output formats ^ and assign error codes to errors via [`Termination`](https://doc.rust-lang.org/std/process/trait.Termination.html).